src/core/datastore/sync/sync-manager.js

```diff
@@ -269,9 +294,29 @@ export class SyncManager {
       .then(() => pushResults);
   }
 
-  _fetchItemsFromServer(collection, query, options) {
-    // TODO: deltaset logic goes here
-    return this._networkRepo.read(collection, query, options);
+  _fetchItemsFromServer(collection, query, options = {}) {
+    return this._networkRepo.read(collection, query, options)
+      .then((data) => {
+        const useDeltaSet = options.useDeltaFetch || false;
+
+        if (useDeltaSet) {
+          return this._getOfflineRepo()
+            .then((repo) => {
+              const { deleted } = data;
+
+              if (isArray(deleted) && deleted.length > 0) {
+                const deletedIds = deleted.map((item) => item._id);
+                const deleteQuery = new Query().containsAll('_id', deletedIds);
```

Comment:
I believe a `$all` filter wouldn't work the same as `$in`, would it? This would never match anything, would it?

Reply:
Yes I meant to use a `$in` filter.